components/tags-filter/TagsFilter.jsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import classnames from 'classnames';
+import FilterBtn from './FilterBtn';
+
+export default function TagsFilter({
+  categorisList,
+  selectedCategories,
+  filterOnChange,
+  pageTitle,
+}) {
+  const disabledBtn = (btn) => {
+    btn.classList.add('-disabled');
+    setTimeout(() => btn.classList.remove('-disabled'), 500);
```

Comment:
@soloviovmax , @antonmorya is there a way to handle this without mutating DOM?

Reply:
Yes
We can modify line 57 like this:
`className={classnames('filter__btn -show-all', { '-active': selectedCategories.length !== categorisList.length }, { '-disabled': selectedCategories.length === categorisList.length }, { '-disabled': selectedCategories.length === 1 })}`
and then just remove lines 39-44, 25-28 and 13-16

Reply:
And yes - we should not check all this conditions right in `className`. Should move it out of `return` method.

Reply:
@antonmorya this class should be added after the click and after 500 ms should be removed.
I deleted this function and made through the state.